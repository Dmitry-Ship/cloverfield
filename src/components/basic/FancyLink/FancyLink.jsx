import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';

import { link } from './FancyLink.scss';

const FancyLink = ({ children, className, ...rest }) => (
  <Link className={`${link} ${className}`} {...rest} >
    <h3 >{children}</h3>
  </Link>
);

FancyLink.defaultProps = {
  className: '',
  to: '#',
};

FancyLink.propTypes = {
  className: PropTypes.string,
  children: PropTypes.any.isRequired,
  to: PropTypes.string,
};

export default FancyLink;
