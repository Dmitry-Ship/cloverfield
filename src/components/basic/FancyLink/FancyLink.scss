@value breakPoints: "../../../styles/breakPoints.scss";
@value mobile from breakPoints;
@value colors: "../../../styles/colors.scss";
@value secondary, tertiary  from colors;


.link {
  margin-top: 3em;
  display: block;
  text-align: center;
  color: tertiary;
  &:hover {
    color: secondary;
  }
  @media mobile {
    font-size: .8rem;
  }
}