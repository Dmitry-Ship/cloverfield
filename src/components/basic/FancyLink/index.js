export { default } from './FancyLink.jsx';
