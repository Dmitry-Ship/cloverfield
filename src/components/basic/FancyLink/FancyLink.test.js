import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import FancyLink from './FancyLink';

describe('<FancyLink />', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <FancyLink />
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render whith children passed in', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <FancyLink>Hello World</FancyLink>
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
