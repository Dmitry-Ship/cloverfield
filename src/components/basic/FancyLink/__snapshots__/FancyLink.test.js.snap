// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FancyLink /> should render correctly 1`] = `
<a
  className="link null"
  href="/"
  onClick={[Function]}
>
  <h3 />
</a>
`;

exports[`<FancyLink /> should render whith children passed in 1`] = `
<a
  className="link null"
  href="/"
  onClick={[Function]}
>
  <h3>
    Hello World
  </h3>
</a>
`;
